src/GraphQL/Execution/ParallelExecutionStrategy.cs

```diff
@@ -14,7 +14,7 @@ protected async Task ExecuteNodeTreeAsync(ExecutionContext context, ExecutionNod
             var pendingNodes = new Queue<ExecutionNode>();
             pendingNodes.Enqueue(rootNode);
 
-            var currentTasks = new List<Task<ExecutionNode>>();
+            var currentTasks = new List<(Task, ExecutionNode)>();
```

Comment:
I would prefer to split this data into two lists. It allows you to eliminate `Select` calls later.

Reply:
Yeah I thought about that. Just causes two object allocations when expanding the array. I can change it

Reply:
`Select`s will also allocate, so...

Reply:
We will see the results, it will be interesting to compare 😄

Reply:
The code in question never runs since none of the async resolvers calls Task.Yield.

Reply:
(During benchmarks)

Reply:
Done.  Benchmarks did not change, of course.